### PR TITLE
feat(module:table): radio selection type

### PR DESCRIPTION
**Pull request by ParsaArvanehPA**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [✔] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [✔] Tests for the changes have been added (for bug fixes / features)
- [✔] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[✔] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
In [Ant design](https://ant.design/components/table#components-table-demo-row-selection) it is possible to have both `checkbox` and `radio` as selection type for columns; meanwhile we only support `checkbox` selection type.

Issue Number: #8029 

## What is the new behavior?
Now supports radio selection type.

## Does this PR introduce a breaking change?
```
[ ] Yes
[✔] No
```

### CODE REVIEW

File: components/table/doc/index.zh-CN.md [hunk 1]
```diff
@@ -116,12 +116,19 @@ Table 组件同时具备了易用性和高度可定制性
 | `[nzChecked]`       | checkbox 是否被选中，可双向绑定 | `boolean`               | -      |
 | `(nzCheckedChange)` | 选中的回调                      | `EventEmitter<boolean>` | -      |
 
+无线电财产
```
Comment (Laffery) [inline components/table/doc/index.zh-CN.md:119]: translation erorr :)

--> 单选属性

**Review by Laffery — CHANGES_REQUESTED** [review]

File: components/table/demo/selection.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+order: 2
```
[the remark this answers is not in the document]
Comment (ParsaArvanehPA) [reply components/table/demo/selection.md:2]: This happens to be the only one with order value of 2; first order is 0, then jumps to 3.

File: components/table/doc/index.en-US.md [hunk 3]
```diff
@@ -164,16 +172,29 @@ Other
...
+| `[nzDisabled]`         | Whether disable checkbox          | `boolean`               | -            |
+| `[nzIndeterminate]`    | Indeterminate status              | `boolean`               | -            |
+| `[nzLabel]`            | ARIA label for the `nz-checkbox`  | `string`                | -            |
+| `[nzChecked]`          | Checked status, double binding    | `boolean`               | -            |
+| `(nzCheckedChange)`    | Checked status change callback    | `EventEmitter<boolean>` | -            |
+| `[nzRowSelectionType]` | Checkbox or radio                 | `'checkbox' \| 'radio'` | `'checkbox'` |
+| `[colSpan]`            | how many columns the cell extends | `number`                | `null`       |
+| `[rowSpan]`            | how many rows the cell extends    | `number`                | `null`       |
+
+radio property
```
[the remark this answers is not in the document]
Comment (ParsaArvanehPA) [reply components/table/doc/index.en-US.md]: fixed.

File: components/table/src/cell/td-addon.component.ts [hunk 4]
```diff
@@ -108,7 +129,11 @@ export class NzTdAddOnComponent implements OnChanges {
       this.nzShowExpand = true;
     }
     if (isFirstChange(nzChecked) && !this.isNzShowCheckboxChanged) {
-      this.nzShowCheckbox = true;
+      if (this.nzRowSelectionType === 'checkbox') {
+        this.nzShowCheckbox = true;
+      } else {
+        this.nzShowRadio = true;
```
[the remark this answers is not in the document]
Comment (ParsaArvanehPA) [reply components/table/src/cell/td-addon.component.ts:122]: yeah you are right, done.

File: components/table/doc/index.zh-CN.md (hunk 1 above)
Comment (ParsaArvanehPA) [reply components/table/doc/index.zh-CN.md]: done :)

**Comment by ParsaArvanehPA** [comment]
Your requested changes are made, could you please check it again?
@Laffery

**Comment by ParsaArvanehPA** [comment]
@Laffery 
Don't know why if failed the second time i pushed last time :)
now is fixed 🙃

**Comment by ParsaArvanehPA** [comment]
closed due to inactivity on my part, re-opened in a new pr

**Pull request closed without merging**